ops/model.py

```diff
@@ -1589,7 +1591,20 @@ def application_version_set(self, version):
         self._run('application-version-set', '--', version)
 
     def juju_log(self, level, message):
-        self._run('juju-log', '--log-level', level, "--", message)
+        # In most cases, simply log.
+        if len(message) <= MAX_STR_LEN:
+            return self._run('juju-log', '--log-level', level, "--", message)
+
+        # Handle very long log messages.
+        messages = [
+            "Log string greater than {}. Splitting into multiple chunks: ".format(MAX_STR_LEN)
+        ]
+        while message:
```

Comment:
Do we need to consider a maximum message length before we just refuse to output to `juju-log` in the case that someone throws something mental at it?